pages/api/admin/CheckIn.js

```diff
@@ -0,0 +1,161 @@
+import { validateFunc } from "../validate";
+import { google } from "googleapis";
+
+import service from "../../../service-account.enc";
+import decrypt from "../../../utils/decrypt.js";
+
+const test = process.env.NEXT_PUBLIC_VERCEL_ENV == undefined;
+const checkin_sheet = test
+  ? process.env.SPREADSHEET_ID_TEST
+  : process.env.PANTRY_DATA_SPREADSHEET_ID;
+
+/*
+ * /api/admin/checkin
+ * req.body = { string calID }
+ */
+
+function requireParams(body, res) {
+  // makes sure that the input is in the right format
+  // returns false and an error if not a good input
+  if (body.calID) return true;
+  return false;
+}
+
+//returns the start of the week given a day
+function determineStartOfWeek(currDay) {
+  let startOfWeek = new Date(currDay.toLocaleDateString());
+  const millisecondsPerDay = 86400000;
+  const milSecondsUntilStart = millisecondsPerDay * currDay.getDay();
+  startOfWeek.setTime(startOfWeek.getTime() - milSecondsUntilStart);
+  return startOfWeek.getTime();
+}
+
+//converts from 2022-07-23T20:35:41.935Z to 7/23/2022 12:15:52
+function formatTime(timeToConvert) {
+  const formattedHours =
+    timeToConvert.getHours() +
+    ":" +
+    timeToConvert.getMinutes() +
+    ":" +
+    timeToConvert.getSeconds();
+  const formattedTime =
+    timeToConvert.toLocaleDateString() + " " + formattedHours;
+  return formattedTime;
+}
```

Comment:
Woahh didn't know you could do this haha

Reply:
changed to this but still uses military time (en-GB instead of en-US since google forms uses it and don't want the data team to have to change their models)